### PR TITLE
Get machine by name

### PR DESCRIPTION
**Pull request by AugustoPeres**
This pull request adds a function that allows retrieval of a single machine group by its name:

```
>>> import inductiva
>>> mg = inductiva.resources.machine_groups.get_by_name("api-6j0zskh4hmcv1wjevnwe2o176")
>>> mg
MachineGroup(name="api-6j0zskh4hmcv1wjevnwe2o176")
```

### CODE REVIEW

File: inductiva/resources/machine_groups.py [hunk 1]
```diff
@@ -118,6 +118,30 @@ def list(fout: TextIO = sys.stdout):
...
 
+def get_by_name(machine_name: str):
+    """Returns the machine group corresponding to `machine_name`."""
+    machine_groups = _fetch_machine_groups_from_api()
+    machine_group = next(
+        (mg for mg in machine_groups if mg["name"] == machine_name), None)
+    if machine_group is not None:
+        mg_class = _get_machine_group_class_from_api_response(machine_group)
+        return mg_class.from_api_response(machine_group)
+    return None
```
Comment (sergiomsantos) [inline inductiva/resources/machine_groups.py:129]: there's already an endpoint for this: `/compute/group/{name}`
You should be able to get the machine directly.

File: inductiva/resources/machine_groups.py [hunk 2]
```diff
@@ -118,6 +118,30 @@ def list(fout: TextIO = sys.stdout):
...
+
+
+def _get_machine_group_class_from_api_response(mg: dict):
```
Comment (sergiomsantos) [inline inductiva/resources/machine_groups.py:132]: make it more explicit. It is easier for testing purposes and decouples it from having to have a dict.
What if the dict doesn't have the required kesy?

```suggestion
def _get_machine_group_class_from_api_response(type: str, is_elastic: bool):
    """Get the suitable machine group class for the given arguments""" 
```

File: inductiva/resources/machine_groups.py [hunk 3]
```diff
@@ -126,14 +150,7 @@ def get():
...
     for mg in machine_groups:
-        if mg["is_elastic"]:
-            mg_class = resources.ElasticMachineGroup
-        elif mg["type"] == "standard":
-            mg_class = resources.MachineGroup
-        elif mg["type"] == "mpi":
-            mg_class = resources.MPICluster
-        else:
-            raise ValueError("Unknown resource configuration.")
+        mg_class = _get_machine_group_class_from_api_response(mg)
```
Comment (sergiomsantos) [inline inductiva/resources/machine_groups.py:153]:
```suggestion
        mg_class = _get_machine_group_class_from_api_response(
                mg["type"], mg["is_elastic"])
```

**Review by sergiomsantos — CHANGES_REQUESTED** [review]
missing unit tests;
a single machine group can now be retrieved directly from the backend.

**Comment by AugustoPeres** [comment]
## QA

Launch every type of machine. Use the function (screen-shot above) to fetch the machine by name

**Review by IvanPombo — CHANGES_REQUESTED** [review]
LGTM! But I am curious how you have tested anything with it, since just by pulling the branch and starting the machine it doesn't work.

The quick fix, is to merge the PR on the ICE client, since I have done the corresponding changes there.

**Comment by AugustoPeres** [comment]
> LGTM! But I am curious how you have tested anything with it, since just by pulling the branch and starting the machine it doesn't work.
> 
> The quick fix, is to merge the PR on the ICE client, since I have done the corresponding changes there.

It fails on launching the machine but not on fetching. I just launched the machines from a different virtual environment

**Comment by AugustoPeres** [comment]
> LGTM! But I am curious how you have tested anything with it, since just by pulling the branch and starting the machine it doesn't work.
> 
> The quick fix, is to merge the PR on the ICE client, since I have done the corresponding changes there.

@ Ivan Pombo, changed base and merged with your branch

**Comment by AugustoPeres** [comment]
*NOTE*: This pull request now also has changes from https://github.com/inductiva/inductiva/pull/1229 and https://github.com/inductiva/inductiva/pull/1218 which where necessary for testing.

They should disappear once those pull requests are merged to dev

**Review by sergiomsantos — APPROVED** [review]

**Comment by Pbarbosa92** [comment]
**QA**
✅ 
I created 3 machine groups and selected each one by name.
Captura de ecrã 2024-03-26, às 12 04 06

**Review by IvanPombo — APPROVED** [review]

**Pull request merged**